internal/envtest/envtest.go

```diff
@@ -48,7 +48,7 @@ type ServerConfigResponse struct {
 // BuildFormRequest builds an http request and http response recorder for the
 // given form values (expressed as url.Values). It sets the proper headers and
 // response types to post as a form and expect HTML in return.
-func BuildFormRequest(ctx context.Context, tb testing.TB, meth, pth string, v *url.Values) (*httptest.ResponseRecorder, *http.Request) {
+func BuildFormRequest(ctx context.Context, tb testing.TB, meth, pth string, v *url.Values, customHeaders *http.Header) (*httptest.ResponseRecorder, *http.Request) {
```

Comment:
I'm not sure I like this change. The builder returns the `http.Request` for this kind of customization on purpose. I would push this into the test itself:

```go
w, r := BuildFormRequest(ctx, t, "GET", "/", nil)
r.Header.Set("foo", "bar")
```